crates/phactory/src/contracts/btc_price_bot.rs

```diff
@@ -0,0 +1,283 @@
+
+                // This Command triggers the use of `AsyncSideTask`, it first send a HTTP request to get the current BTC
+                // price from https://min-api.cryptocompare.com/, then sends the price to a Telegram bot through another
+                // HTTP request
+                //
+                // To ensure the state consistency, the time to start the task and the time to upload the HTTP response
+                // to chain must be determined. In this case, we start the task in the current `block_number`, and report
+                // the result, whether succeeded or failed, to the chain after `duration`
+                //
+                // Report the result after 2 blocks no matter whether has received the HTTP response
+                let block_number = context.block.block_number;
+                let duration = 2;
+
+                let mq = context.mq().clone();
+                let my_id = context.self_id;
+
+                let default_messages = [mq.prepare_message_to(&(), "^phala/mq/blockhole")];
+
+                context.block.side_task_man.add_async_task(
+                    block_number,
+                    duration,
+                    default_messages,
+                    async move {
+                        // Do network request in this block and return the result.
+                        // Do NOT send mq message in this block.
+                        log::info!("Side task starts to get BTC price");
+                        let mut resp = match surf::get(
+                            "https://min-api.cryptocompare.com/data/price?fsym=BTC&tsyms=USD",
+                        )
+                        .send()
+                        .await
+                        {
+                            Ok(r) => r,
+                            Err(_err) => {
+                                return Err(Error::NetworkUnavailable.into());
+                            }
+                        };
+                        let result = match resp.body_string().await {
+                            Ok(body) => body,
+                            Err(err) => {
+                                format!("Network error: {:?}", err)
+                            }
+                        };
+                        log::info!("Side task got BTC price: {}", result);
+
+                        let price: BtcPrice =
+                            serde_json::from_str(result.as_str()).or(Err(Error::BadBtcPrice))?;
+                        let text = format!("BTC price: ${}", price.usd);
+                        let uri = format!(
+                            "https://api.telegram.org/bot{}/{}",
+                            bot_token, "sendMessage"
+                        );
+                        let data = &TgMessage { chat_id, text };
+
+                        let mut resp = match surf::post(uri)
+                            .body_json(data)
+                            .expect("should not fail with valid data; qed.")
```

Comment:
What will happen if the result is not a json? Will the worker crash as a whole?

Reply:
return an error instead of panic